packages/python/plotly/plotly/express/_core.py

```diff
@@ -754,6 +754,210 @@ def apply_default_cascade(args):
+                    "at the moment." % field
+                )
+            ## ----------------- argument is a col name ----------------------
+            if isinstance(argument, str) or isinstance(
+                argument, int
+            ):  # just a column name given as str or int
+                if not df_provided:
+                    raise ValueError(
+                        "String or int arguments are only possible when a"
+                        "DataFrame or an array is provided in the `data_frame`"
+                        "argument. No DataFrame was provided, but argument '%s'"
+                        "is of type str or int." % field
+                    )
+                # Check validity of column name
+                if argument not in df_input.columns:
+                    err_msg = (
+                        "Value of '%s' is not the name of a column in 'data_frame'. "
+                        "Expected one of %s but received: %s"
+                        % (field, str(list(df_input.columns)), argument)
+                    )
+                    if argument == "index":
+                        err_msg += (
+                            "\n To use the index, pass it in directly as `df.index`."
+                        )
+                    raise ValueError(err_msg)
+                if length and len(df_input[argument]) != length:
+                    raise ValueError(
+                        "All arguments should have the same length. "
+                        "The length of column argument `df[%s]` is %d, whereas the "
+                        "length of previous arguments %s is %d"
+                        % (
+                            field,
+                            len(df_input[argument]),
+                            str(list(df_output.columns)),
+                            length,
+                        )
+                    )
+                col_name = str(argument)
+                df_output[col_name] = df_input[argument]
+            # ----------------- argument is a column / array / list.... -------
+            else:
+                is_index = isinstance(argument, pd.RangeIndex)
+                # First pandas
+                # pandas series have a name but it's None
+                if (
+                    hasattr(argument, "name") and argument.name is not None
+                ) or is_index:
+                    col_name = argument.name  # pandas df
+                    if col_name is None and is_index:
+                        col_name = "index"
+                    if not df_provided:
+                        col_name = field
+                    else:
+                        if is_index:
+                            keep_name = df_provided and argument is df_input.index
+                        else:
+                            # we use getattr/hasattr because of index
+                            keep_name = hasattr(
```

Comment:
```python
df = px.data.tips()
px.scatter(df, x=df["size"], y=df.tip)
```

This has a weird behaviour: it doesn't pick up the name "size" even though it is a valid column... This is almost certainly because `df.size` doesn't return `df["size"]`. Thanks Pandas, again :(

Reply:
I think the solution is to look up columns via `df_input["name"]` and not `getattr` and just special-case the `index` thing.

Reply:
actually it was already a special case, so the modification was easy